### ISSUE TITLE
refactor: ModelConfig — abstract input_names/output_name/pooling for non-BERT models

### ISSUE THREAD

**Issue by jamie8johnson** [jamie8johnson/cqs#949]
## Problem

The ONNX embedder (`src/embedder/mod.rs`) hardcodes BERT-style assumptions:

- Input tensor names: `input_ids`, `attention_mask`, `token_type_ids` — assumed.
- Output tensor name: `last_hidden_state` (or sometimes `sentence_embedding`) — assumed.
- Pooling: mean-pooling over the masked token positions — hardcoded.
- Tokenizer: BERT WordPiece via `tokenizers` crate — assumed via the BGE / E5 model presets.

Adding a model that diverges on any of these (different I/O names, different pooling, BPE tokenizer) is painful — requires editing the encoder code, not just adding a config entry.

The audit (EX-V1.25-6) flagged this as the limiting factor for swapping in non-BERT-family models. Particularly relevant given:

- **Switching the production default from BGE-large to E5 v9-200k** is on the roadmap (today's eval showed they tie on quality, E5 is 1/3 the size).
- **Future model experiments** (Jina v3, Stella, GTE, custom-trained) all want a clean way to be added without forking the encoder.

## Proposal

Extend `ModelConfig` to describe the model architecture, not just point at it:

```rust
#[derive(Debug, Clone, Deserialize)]
pub struct ModelConfig {
    pub source: ModelSource,    // hf hub repo or local path
    pub revision: Option<String>,
    pub dim: usize,
    pub max_seq_len: usize,

    // NEW
    pub input_names: InputNames,
    pub output_name: String,
    pub pooling: PoolingStrategy,
    pub tokenizer: TokenizerKind,
}

#[derive(Debug, Clone, Deserialize)]
pub struct InputNames {
    pub ids: String,             // default "input_ids"
    pub mask: String,            // default "attention_mask"
    pub token_types: Option<String>,  // None for models without token_type_ids
}

#[derive(Debug, Clone, Deserialize, PartialEq)]
pub enum PoolingStrategy {
    Mean,        // current default
    Cls,         // first-token
    LastToken,   // some autoregressive embedders
}

#[derive(Debug, Clone, Deserialize)]
pub enum TokenizerKind {
    WordPiece(Box<tokenizers::Tokenizer>),
    Bpe(Box<tokenizers::Tokenizer>),
    SentencePiece(Box<tokenizers::Tokenizer>),
}
```

Then the encoder reads from `config` instead of hardcoding:

```rust
let inputs = ort::inputs![
    config.input_names.ids.as_str() => ids_array,
    config.input_names.mask.as_str() => mask_array,
];
// optional token_types if Some
let outputs = session.run(inputs)?;
let pooled = match config.pooling {
    PoolingStrategy::Mean => mean_pool(&outputs[config.output_name.as_str()], &mask),
    PoolingStrategy::Cls => cls_pool(&outputs[config.output_name.as_str()]),
    PoolingStrategy::LastToken => last_token_pool(&outputs[config.output_name.as_str()], &mask),
};
```

Existing model presets (BGE-large, E5-base, v9-200k) get explicit configs in `src/embedder/presets.rs`. They are unchanged in behavior — just declared instead of implicit.

## Impact

- **Adding a new model becomes a config entry, not a code edit.**
- Unblocks the BGE → E5 v9-200k default switch by making swap testing easy.
- Future-proofs against non-BERT embedders (LLM-based encoders, Stella, etc.).
- Audit finding EX-V1.25-6 closed.
- Related to EX-V1.25-7 (SPLADE/reranker model preset registry) — same idea applied to those two model classes.

## Acceptance criteria

- [ ] `ModelConfig` carries input_names, output_name, pooling, tokenizer.
- [ ] Existing presets (BGE-large, E5-base, v9-200k, custom ONNX) all expressed as configs — no behavior change.
- [ ] At least one new model preset added that proves the abstraction works for a non-BERT-default model (e.g., a CLS-pooled model — even if it's just for testing).
- [ ] Existing eval (265q V2, BGE-large default) produces the same R@1 before and after the refactor (regression check).
- [ ] Documentation: README "How It Works" section describes how to add a new model preset.

## Cross-references

- Audit triage: `docs/audit-triage.md` § P4 GitHub issues → EX-V1.25-6
- Related: EX-V1.25-7 (SPLADE/reranker preset registry) — companion refactor for those model classes
- Pre-requisite for: E5 v9-200k as production default (today's eval result)
- Roadmap: "Refactoring Lane" entry added 2026-04-14
